patches/net/minecraft/world/effect/MobEffect.java.patch

```diff
@@ -9,6 +9,19 @@
      public static final Codec<Holder<MobEffect>> CODEC = BuiltInRegistries.MOB_EFFECT.holderByNameCodec();
      public static final StreamCodec<RegistryFriendlyByteBuf, Holder<MobEffect>> STREAM_CODEC = ByteBufCodecs.holderRegistry(Registries.MOB_EFFECT);
      private static final int AMBIENT_ALPHA = Mth.floor(38.25F);
+@@ -64,6 +_,12 @@
+         this.particleFactory = p_333515_ -> p_333716_;
+     }
+ 
++    protected MobEffect(MobEffectCategory category, int color, Function<MobEffectInstance, ParticleOptions> particleFactory) {
```

Comment:
We should add a javadoc comment to this constructor, not necessarily to inform the users of the parameters (they're pretty self-explanatory), but do to the following:
1. Inform the user this is a patched-in ctor (i.e. add a `Neo:` header)
2. Inform the user that they should use this constructor over the `ParticleOptions` one if using a `ParticleOptions` that depends on other registry entries.